homeassistant/helpers/service.py

```diff
@@ -113,3 +117,58 @@ def extract_entity_ids(hass, service_call):
         return group.expand_entity_ids(hass, [service_ent_id])
 
     return [ent_id for ent_id in group.expand_entity_ids(hass, service_ent_id)]
+
+
+# pylint: disable=too-few-public-methods
+class CallRateDelayThrottle(object):
```

Comment:
OK.  I'll put throttling  the code directly in the component as soon I've time and try to play around with async in a seperate branch.  I'll tell you if I've done any significant progress.  Depends all a bit on how much time I have.